pkg/config/app_config.go

```diff
@@ -341,6 +346,41 @@ func changeCommitPrefixesMap(rootNode *yaml.Node) error {
 	})
 }
 
+func changeCustomCommandStreamAndOutputToOutputEnum(rootNode *yaml.Node) error {
+	return yaml_utils.Walk(rootNode, func(node *yaml.Node, path string) {
+		// We are being lazy here and rely on the fact that the only mapping
+		// nodes in the tree under customCommands are actual custom commands. If
+		// this ever changes (e.g. because we add a struct field to
+		// customCommand), then we need to change this to iterate properly.
+		if strings.HasPrefix(path, "customCommands[") && node.Kind == yaml.MappingNode {
+			output := ""
+			if streamKey, streamValue := yaml_utils.RemoveKey(node, "stream"); streamKey != nil {
+				if streamValue.Kind == yaml.ScalarNode && streamValue.Value == "true" {
+					output = "log"
```

Comment:
In the interest of backwards compatibility, what are your thoughts on migrating everybody who's currently on 'stream: true' to both 'output: log' and 'pty: true'?

Reply:
My assumption is that people using `stream: true` did that because they wanted to see the output, not because they wanted to use a PTY (like me, when I discovered the issue). I think it makes sense to err on the side of what's probably best for most people, rather than what faithfully replicates the previous behavior.

Reply:
But how do we actually know what's best for most people? The fact that we've come this far without anybody complaining suggests that the current way actually is the best way.

Reply:
I did not mean to imply that you were not yourself complaining, my point was that we've come this far without any complaints prior, meaning lots of people may be happy with the current default.

Nevertheless you've persuaded me: it's much more painful to realise that you _don't_ need a pty than vice versa, and it's possible for an issue to arise out of nowhere like it did with your gh.